snapshot/restore/local_fs_output.go

```diff
@@ -402,16 +430,20 @@ func (o *FilesystemOutput) copyFileContent(ctx context.Context, targetPath strin
 		return errors.Wrap(err, "unable to open snapshot file for "+targetPath)
 	}
 	defer r.Close() //nolint:errcheck
+	wr := &progressReportingReader{
+		r:  r,
+		cb: progressCb,
+	}
 
 	log(ctx).Debugf("copying file contents to: %v", targetPath)
 	targetPath = atomicfile.MaybePrefixLongFilenameOnWindows(targetPath)
 
 	if o.WriteFilesAtomically {
 		//nolint:wrapcheck
-		return atomicfile.Write(targetPath, r)
+		return atomicfile.Write(targetPath, wr)
 	}
 
-	return write(targetPath, r, f.Size(), o.copier)
+	return write(targetPath, wr, f.Size(), o.copier)
```

Comment:
That's true, but this way of progress update was chosen because it is less intrusive. Also, when reading, usually, the block is very small in comparison with total amount of data. So, we, anyway, can get 100% earlier than last block will be read. So, I'd not consider this as critical flaw.